src/System.Runtime.InteropServices/tests/System/Runtime/InteropServices/Marshal/Miscellaneous/MarshalClassTests.cs

```diff
@@ -0,0 +1,188 @@
+#pragma warning disable 649
+        //definition of structure that will be used in testing of structs with Fixed BSTR Safearray fields
+        internal struct Variant
+        {
+            public ushort vt;
+            public ushort wReserved1;
+            public ushort wReserved2;
+            public ushort wReserved3;
+            public IntPtr bstrVal;
+            public IntPtr pRecInfo;
+        }
+#pragma warning restore 649
+
+        [StructLayout(LayoutKind.Auto)]
+        public struct SomeTestStruct_Auto
+        {
+            public int i;
+        }
+
+        [StructLayout(LayoutKind.Sequential)]
+        public struct StructWithFxdLPSTRSAFld
+        {
+            [MarshalAs(UnmanagedType.ByValArray, ArraySubType = UnmanagedType.LPStr, SizeConst = 0)]
+            public String[] Arr;
+        }
+
+        [StructLayout(LayoutKind.Sequential)]
+        public struct SomeTestStruct
+        {
+            public int i;
+            //[MarshalAs(UnmanagedType.BStr)]
+            public String s;
+        }
+
+        public enum TestEnum
+        {
+            red,
+            green,
+            blue
+        }
+
+        public struct TestStructWithEnumArray
+        {
+            [MarshalAs(UnmanagedType.ByValArray, SizeConst = 3)]
+            public TestEnum[] ArrayOfEnum;
+        }
+
+        public IntPtr ip;
+        public SomeTestStruct someTs = new SomeTestStruct();
+        public Exception ex = null;
+
+        [STAThread]
+        [Fact]
+        public void StructWithEnumArray()
+        {
+            TestStructWithEnumArray s = new TestStructWithEnumArray();
+            s.ArrayOfEnum = new TestEnum[3];
+            s.ArrayOfEnum[0] = TestEnum.red;
+            s.ArrayOfEnum[1] = TestEnum.green;
+            s.ArrayOfEnum[2] = TestEnum.blue;
+            int retsize = Marshal.SizeOf(s.GetType());
+            Assert.Equal(12, retsize);
+
+            retsize = 0;
+            retsize = Marshal.SizeOf(typeof(TestStructWithEnumArray));
+            int genericRetsize = Marshal.SizeOf<TestStructWithEnumArray>();
+            Assert.True(genericRetsize == retsize, "Generic and non generic versions of the API did not return the same size!");
+            Assert.Equal(12, retsize);
+        }
+
+        [STAThread]
+        [Fact]
+        public void StructureToPtr()
+        {
+            ip = IntPtr.Zero;
+            Assert.Throws<ArgumentNullException>("ptr", () => Marshal.StructureToPtr<SomeTestStruct>(someTs, ip, true));
+
+            ip = new IntPtr(123);
+            Assert.Throws<ArgumentNullException>("structure", () => Marshal.StructureToPtr<Object>(null, ip, true));
+
+            ip = Marshal.AllocHGlobal(Marshal.SizeOf(someTs));
+            someTs.s = "something";
+            ex = Record.Exception(() => Marshal.StructureToPtr(someTs, ip, false));
+            Assert.Null(ex);
+            ex = Record.Exception(() => Marshal.StructureToPtr(someTs, ip, true));
+            Assert.Null(ex);
```

Comment:
This pattern is just validating that the calls don't throw exceptions.  As such, you don't need any of this xunit stuff and can just call the methods: if they throw an exception, it'll propagate and fail the test:
```C#
Marshal.StructureToPtr(someTs, ip, false);
Marshal.StructureToPtr(someTs, ip, true);
```

Reply:
Fixed, see in next commit